Mejoras completas en registro, inscripción, navegación, administración y seguridad

Andriu-Dex opened the pull request:
Esta Pull Request integra múltiples mejoras visuales, funcionales y estructurales en el sistema **AcademicEvents**, cubriendo tanto la experiencia del usuario estudiante como la del administrador.

---

#### 🎨 **INTERFAZ Y EXPERIENCIA DE USUARIO**

* Rediseño del formulario de registro con estilo moderno, íconos `lucide-react`, feedback visual, validaciones estrictas y uso de `select` dinámico para carreras.
* Validaciones integradas:

  * Cédula: exactamente 10 dígitos
  * Nombres y apellidos: solo letras
  * Carreras: opción seleccionable desde la base de datos
* Navegación protegida y diferenciada por roles:

  * Estudiantes redirigidos a `/eventos`
  * Administradores redirigidos a `/admin`
* Protección robusta de rutas administrativas con `PrivateRouteAdmin`.

---

#### 🧠 **ADMINISTRACIÓN DE CARRERAS Y CONFIGURACIÓN**

* **/admin/carreras**

  * Crear, editar y eliminar carreras con confirmación
* **/admin/configuracion**

  * Gestión de misión, visión y autoridades institucionales desde la interfaz
  * Cambios persistentes en la base de datos (`configuracion` con `id_conf = 1`)

---

#### 📋 **GESTIÓN DE INSCRIPCIONES – Admin**

* Panel completo en `/admin/inscripciones` con:

  * Filtro por evento
  * Visualización de nombre, correo, comprobante (PDF o imagen), nota, asistencia
  * Botones para aceptar, rechazar o finalizar inscripciones
* Finalización de cursos:

  * Valida que se haya ingresado `asistencia` y `nota_final`
  * Verifica si se cumple mínimo: `nota ≥ 8`, `asistencia ≥ 80%`
  * Muestra errores si no cumple requisitos

---

#### 🔐 **SEGURIDAD Y BACKEND**

* Corrección en la generación y validación del `JWT`:

  * `rol_usu` ahora se incluye correctamente en el token y en la respuesta
* Rutas reorganizadas:

  * Todas las rutas de inscripciones protegidas y organizadas
  * Rutas `admin/inscripciones/validar/:id` y `admin/inscripciones/evento/:id` funcionando con middleware de autorización
* Manejo de expiración del token pendiente de mejora futura (actualmente no muestra alerta)

---

#### 🧪 **PRUEBAS RECOMENDADAS**
* Crear un archivo `.env` dentro del frontend con la línea: 

```env
VITE_API_URL=http://localhost:3000
```
O el puerto que se utilice.
(Facilita los cambios de entorno sin modificar el código).

* Dentro de la carpeta backend olocar los comandos:
```bash
`npx prisma db push
npm run seed`
```
* Crear un usuario con el rol ADMIN desde la base de datos:
```sql
`INSERT INTO usuario (
    id_usu, ced_usu, nom_usu, ape_usu, cor,_usu, con_usu, cel_usu, rol_usu, fec_cre_usu, comprobante, carrera
) VALUES (
    'uuid',                   -- ID único en formato UUID
    'cedula',                -- Número de cédula del usuario
    'nombres',               -- Nombres del usuario
    'apellidos',             -- Apellidos del usuario
    'correo_electronico',    -- Correo electrónico institucional o personal
    'contrasena_encriptada', -- Contraseña encriptada (hash bcrypt u otro)
    'numero_celular',        -- Número de celular del usuario
    'ADMIN',                 -- Rol del usuario (por ejemplo: ADMIN, ESTUDIANTE, DOCENTE)
    NOW(),                   -- Fecha y hora de creación automática
    NULL,                    -- Carrera (puede ser NULL para ADMIN)
    NULL                     -- Archivo adjunto (puede ser NULL si no aplica)
);
`
```

* Probar:
  * Registro y login
  * Inscripción a eventos con carga de comprobante
  * Verificación de comprobantes

Iniciar sesión como ADMIN
* Probar:
  * Creación de carreras y edición de configuración institucional
  * Visualización y validación de inscripciones

---

#### ✅ **Checklist final**

* [x] Validaciones frontend y backend funcionando correctamente
* [x] Flujo de login con roles funcionando y protegidos
* [x] Administración de carreras y configuración completa
* [x] Finalización de inscripciones con lógica validada
* [x] Visualización de comprobantes en admin
* [x] Rutas reorganizadas y protegidas
* [x] Sin logs de consola ni errores visibles

---
Closes #30

Andriu-Dex committed:
Agrega vista MyInscriptions con reenvío de comprobantes, descarga de certificados, validaciones, modal reutilizable y lógica protegida con token

Andriu-Dex committed:
Implementación de AdminEvents: Uso de useCallback, manejo de errores con toast, optimización de fechas, accesibilidad en botones y estilo consistente con Tailwind

Andriu-Dex committed:
Protección de ruta /admin/inscripciones/:id con middleware onlyAdmin

Andriu-Dex committed:
Refactor: unificación de rutas administrativas en admin.routes.js

Andriu-Dex committed:
Mejoras en gestión de eventos e inscripciones:
- Corrección en validación de login (se permite cualquier dominio para público general)
- Visualización del nombre del evento en inscripciones de administrador
- Nuevo estilo para botones de filtro de estado
- Reestructuración completa del archivo evento.routes.js
- Mejora en mensajes de error en backend y control de fechas inválidas

Andriu-Dex committed:
Agregar ruta para obtener inscripciones por usuario y reparación de inonsistencias en recepción del token

Andriu-Dex committed:
Registro validado: cedula con 10 dígitos, nombres solo letras; selección de carrera dinámica desde backend; gestión completa de carreras en panel admin

Andriu-Dex committed:
Protección de rutas con PrivateRouteAdmin para acceso exclusivo de usuarios ADMIN

Andriu-Dex committed:
Configuración institucional y gestión de inscripciones integradas:
- Admin puede editar misión, visión y autoridades desde el panel (protegido por rol)
- Validación de inscripciones por evento habilitada con control de estado y rutas protegidas

Andriu-Dex committed:
Fix: alineación de campo 'rol_usu' entre backend y frontend en respuesta de login: Corrige la desincronización entre el backend y el frontend donde el token JWT y el JSON de respuesta enviaban el campo 'rol' en lugar de 'rol_usu'.

Andriu-Dex committed:
Corrección en redireccion de rutas para el administrador

titoma1500 reviewed (approved):
Buen cambio a la interfaz seguir así y mejorar visualmente el resto de interfaces

M4yb33 reviewed (approved):
Buena mejora, sería bueno organizar mejor los componentes para mantener el código limpio a futuro.

Andriu-Dex committed:
Correción De Errores En Inscripcionés para el estudiante: Error 500 al enviar inscripción y Error 404 al consultar si el estudiante ya está inscrito

Andriu-Dex committed:
Implementacion de la barra de navegacion por rol para todas las vistas necesarias

Andriu-Dex committed:
Funcionalidades completas para inscripción, visualización y certificados

- Finalizada la lógica de inscripción a eventos con comprobante (PDF o imagen)
- Vista de inscripciones del estudiante con estados y reenviar comprobante
- Validación del token para obtener inscripciones propias
- Generación y descarga de certificados para eventos finalizados
- Barra de navegación integrada con rutas protegidas
- Corrección de rutas 404 y autorización con middleware

Andriu-Dex committed:
Corrección de errores en funcionalidad de administración: validación de inscripciones, acceso a comprobantes, y finalización de cursos con nota y asistencia

Gabriel-Spartan reviewed (approved):
Bien implementado y funcional todo al igual que la verificación de todas las funcionalidades de forma correcta
